6_nodejs/week2/index.js

```diff
@@ -0,0 +1,94 @@
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+const { readFileSync } = require("fs");
+const { parse } = require('path');
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/", (req, res) => {
+    res.send("This is a search engine");
+});
+
+app.get("/search", (req, res) => {
+    const query = req.query.q
+    const data = readFileSync('./documents.json');
+    const parsedData = JSON.parse(data);
+
+    if (!query) {
+        res.send(parsedData)
+    }
+
+    for (const document of parsedData) {
+        for (const property in document) {
+            const value = document[property]
+            if (!isNaN(value)) {
+                continue
+            }
+            if (value.toLowerCase().includes(query.toLowerCase())) {
+                res.send(document);
+            }
+        }
+    }
+
+});
+
+app.get("/documents/:id", (req, res) => {
+    const id = req.params.id
+    const data = readFileSync('./documents.json');
+    const parsedData = JSON.parse(data);
+
+    for (const document of parsedData) {
```

Comment:
You could also have used .**filter** here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter